automation/update-from-application-services.py

```diff
@@ -21,6 +23,33 @@
 def main():
     args = parse_args()
     version = calc_version(args)
+
+    if args.create_pr:
+        create_pr_with_updated_source(version)
+    else:
+        update_source(version)
+
+def create_pr_with_updated_source(version):
+    g = github.Github(os.environ['GITHUB_TOKEN'])
+    branch = f"update-as-{version}"
+    if branch_exists(branch):
+        print(f"branch {branch} already exists, quitting")
```

Comment:
nit: What happens if the branch doesn't exist, but `main` is already on the latest - I'm assuming the `git commit` won't commit anything, so the push will push a branch identical to the main branch?

I would assume the PR creation might fail (or create an empty PR? not sure how github handles that)

but we could end up with an orphan branch of some sort - it's not a big deal but it might nicer to quit early with a message similar to this!

Reply:
I want to add that this probably will only happen if the "real" PR is landed fairly quickly than another cron job ran, as we only run the cron job one hour in the day.

Reply:
This comment made me realize there's a way better way to do this.

AFAICT, there's no point in creating a PR since we don't run CI in this repo and I don't think it makes sense to anyways.  We run CI in app-services before we publish the nightly and we run CI in firefox-ios after taking the nightly.  I don't see a reason for CI in the middle.

This means the script can be much simpler.  Just push the commit to `main` and tag it.  This also addresses your concern, since the tags won't be deleted.

What do you think of the new approach?

Reply:
Big +100 to the new approach. Lets just push to main and tag. This removes friction. We might need to:
- Double check if the default github token for github actions has permissions to do that, otherwise we can set a [GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets) with a personal access token, ideally a personal access token of @appsvc-moz (@mhammond has admin access)
- Remove the branch protection rule for main for users with write permission. I believe I already added @appsvc-moz to have write permission to this repo, but we can double check

Reply:
Yes the new approach is definitely the way to go. We want this repo to be as low maintenance as possible and really just be a reflection of a-s in the SPM world!